### ISSUE TITLE
PluginWrapper usage

### ISSUE THREAD

**Issue by eladtamary** [decebals/pf4j#50]
Hi, 
In the demo application you have two kinds of plugins - external plugins located in plugin directory (WelcomePlugin) and default implementation located in the application jar (WhazzupGreeting).
I saw you added it not long ago and this is very good improvement.

The external plugins have a wrapper class that extends Plugin class and annotated with @Extension while default implementation is only annotated with @Extension.

What is the required behavior? Can I have external plugin that is only annotated with @Extension without wrapper class?

Thanks for the help,
Elad Tamary

**Comment by decebals** [comment]
An `Extension` has as parent a plugin(wrapper). Starting with version 0.9 you can define an extension directly in the application jar so this extension has no plugin as a parent, it's not loaded by a `PluginClassLoader`, on short, it's a particular type of extension.
Can you explain your problem related to the current implementation? Why do you want to eliminate the `Plugin` concept and to use only the `Extension` concept?

**Comment by eladtamary** [comment]
Thanks for the quick reply.
I just wanted to know what is the correct way to write the plugin class.

I struggled a little bit at first due to dependencies jars that were mistakenly copied to the <plugin-dir>/lib  by the assembly plugin. This caused the plugins not to be loaded on startup, due to duplicate extension classes.
I added useTransitiveDependencies=false to the assembly dependencySet and it solved the issue.

The logs were great and really helped me understand the issue.

Thanks,
Elad
